src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```
@@ -0,0 +1,254 @@
+            {
+                GetMoreBytesFromStream();
+            }
+            return wasSkipped;
+        }
+
+        internal string ReadNextTokenAsString()
+        {
+            ThrowExceptionIfCompleted();
+
+            if (Read())
+            {
+                return _reader.ReadTokenAsString();
+            }
+
+            return null;
+        }
+
+        internal string GetCurrentBufferAsString()
+        {
+            ThrowExceptionIfCompleted();
+
+            return Encoding.UTF8.GetString(_buffer);
+        }
+
+        internal IList<string> ReadStringArrayAsIList(IList<string> strings = null)
+        {
+            if (TokenType == JsonTokenType.StartArray)
+            {
+                while (Read() && TokenType != JsonTokenType.EndArray)
+                {
+                    string value = _reader.ReadTokenAsString();
+
+                    strings = strings ?? new List<string>();
+
+                    strings.Add(value);
+                }
+            }
+            return strings;
+        }
+
+        internal List<string> ReadStringArrayAsList()
+        {
+            return (List<string>)ReadStringArrayAsIList();
+        }
+
+        internal IReadOnlyList<string> ReadDelimitedString()
+        {
+            ThrowExceptionIfCompleted();
+
+            if (Read())
+            {
+                switch (TokenType)
+                {
+                    case JsonTokenType.String:
+                        var value = GetString();
+
+                        return value.Split(DelimitedStringDelimiters, StringSplitOptions.RemoveEmptyEntries);
+
+                    default:
+                        var invalidCastException = new InvalidCastException();
+                        throw new JsonException(invalidCastException.Message, invalidCastException);
+                }
+            }
+
+            return null;
+        }
+
+        internal bool ReadNextTokenAsBoolOrFalse()
```

This method makes me sad. We control both the reading and writing of the assets file, so I don't understand why we'd have code that would allow a property that we expect to be a boolean to actually be a number or an array or object or whatever, and then silently ignore the schema error.

Since the bigger PR was split, I can no longer "see the forest for the trees", making it hard to find if this method is being used for a legitimate reason 😞

---

It's used a bunch of times in the current implementation of [JsonPackageSpecReader](https://github.com/NuGet/NuGet.Client/blob/1942c17866695ae1175ec3d8b42a6b41bcd6a9c9/src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs#L387C110-L387C110). So least it was being used before, whether or not it's legit idk, but we can look at that during the next PR

---

My instinct immediately was whether it's something that was used during the project.json times, but based on where it's being used, I think it's always used as a bool, but I spent like 30s looking :D